makefiles/arch/riscv.inc.mk

```diff
@@ -38,23 +38,23 @@ ifeq (,$(TARGET_ARCH))
 endif
 
 ifeq ($(TOOLCHAIN),gnu)
-  NEW_RISCV_ISA := $(shell echo "" | $(TARGET_ARCH)-as -march=rv32imac_zicsr -mabi=ilp32 - > /dev/null 2>&1 && echo 1 || echo 0)
+  GCC_DEFAULTS_TO_NEW_RISCV_ISA ?= $(shell echo "typedef int dont_be_pedantic;" | $(TARGET_ARCH)-gcc -march=rv32imac -mabi=ilp32 -misa-spec=2.2 -E - > /dev/null 2>&1 && echo 1 || echo 0)
 endif
```

Comment:
Is this check really needed? I think older GCC version will accept `-misa-spec` but ignore it, no? If so, we could just add `-misa-spec=2.2` unconditionally.

Reply:
At least the legacy `riscv-none-embed-gcc` 10.2.0 used in `riot/riotbuild` does accept this fine, so I guess yes. Let's try that and see if someone complains :)

Reply:
Unfortunately, it seems that I was wrong because with my `riscv32-unknown-elf-gcc (GCC) 10.2.0` compiled from [riscv-gnu-toolchain](https://github.com/riscv-collab/riscv-gnu-toolchain) GCC does indeed complain about `error: unrecognized command-line option '-misa-spec=2.2'` :(

So I guess we do need the check, sorry for the noise.

Reply:
Thanks for verifying!